pkg/catalog/configmeta.go

```diff
@@ -384,13 +385,14 @@ func clusterBaseConfig(
 	kdConfigMaps, cmErr := genConfigConnections(cr)
 	kdSecrets, secErr := genSecretConnections(cr)
 
-	if cmErr != nil {
+	if !errors.IsNotFound(cmErr) && cmErr != nil {
```

Comment:
This looks like it will bail out of gathering all connections info if any of the connected objects are missing. Am I reading that right?

Reply:
If connection obj is missing then we keep going but if it's present BUT non-retrievable then we return. 
Hmmm now thinking again, I think we should never return, the values would be correctly set to nil if things were bad.

Reply:
I'm also looking inside the functions like genClusterConnections -- looping over objects but immediately returning out of the loop if any error.

Reply:
yea, so our strategy should be if any obj is not retrievable then ignore and keep going? For all obj (cluster, cm, and secrets )

Reply:
I think so yes, especially in the not-found case. Skip the missing ones and just populate with everything else.

In the case where we actually get an API error (other than not-found) trying to retrieve something, that's arguably different. In that case I think we should probably not update any connections info at all on this handler pass and just try again next pass.